libsrc/c64/crt0.s

```diff
@@ -98,14 +85,51 @@ L2:     lda     zpsave,x
 
         rts
 
+
 ; ------------------------------------------------------------------------
-; Data
 
-.segment        "ZPSAVE"
+.segment        "INIT"
 
-zpsave: .res    zpspace
+initstart:
 
-.bss
+; Save the zero-page locations that we need.
+
+        ldx     #zpspace-1
+L1:     lda     sp,x
+        sta     zpsave,x
+        dex
+        bpl     L1
+
+; Clear the BSS data.
+
+        jsr     zerobss
+
+; Set up the stack.
+
+        lda     #<(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
+        ldx     #>(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
+        sta     sp
+        stx     sp+1            ; Set argument stack ptr
+
+; Call the module constructors.
+
+        jsr     initlib
+
+; Push the command-line arguments; and, call main().
+
+        jmp     callmain
+
+
+; ------------------------------------------------------------------------
+; Data
+
+.data
 
-spsave: .res    1
 mmusave:.res    1
+spsave: .res    1
```

Comment:
> I guess we agree that we don't like `.res` in data segments.

We don't agree about that.  Look at the end of https://github.com/cc65/cc65/blob/master/libsrc/apple2/mainargs.s
`argv` is defined as

```gas
.data
argv:   .addr   FNAM
        .res    MAXARGS * 2
```

The `.res` in the third line is a sign to readers that the initial values in those locations are irrelevant.  Those variables become important only after some piece of code has put something into them.

The value in `move_init` is important even before the program runs.
But, `mmusave` and `spsave` are significant only after registers are saved into them.  I think that we should use `.res` to show that.